### PR TITLE
Add support for ed25519 private keys

### PR DESCRIPTION
**Pull request by jborean93**
Adds support for using ed25519 private key in user authentication. As .NET does not support ED25519 in the BCL it uses
BouncyCastle.Cryptography as a dependency for the key signing tasks.

Fixes https://github.com/tmds/Tmds.Ssh/issues/204

### CODE REVIEW

File: src/Tmds.Ssh/Ed25519PrivateKey.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace Tmds.Ssh;
+
+sealed class Ed25519PrivateKey : PrivateKey
+{
+    private const int _privateKeySize = 32;
+
+    // Contains the private and public key as one block of bytes from the
+    // serialized OpenSSH key data.
+    private readonly byte[] _keyData;
```
Comment (jborean93) [inline src/Tmds.Ssh/Ed25519PrivateKey.cs:16]: I used a single `byte[]` as that's how it is encoded in the raw OpenSSH key and the BouncyCastle API accepts an offset for the private and public key. Happy to just split that out in the parser and store it as 2 separate arrays if you prefer that.

Comment (tmds) [reply src/Tmds.Ssh/Ed25519PrivateKey.cs:16]: Yes, please split into separate arrays for the private and public key.

File: src/Tmds.Ssh/Ed25519PrivateKey.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+using System;
+using System.Buffers;
+using Org.BouncyCastle.Math.EC.Rfc8032;
+
+namespace Tmds.Ssh;
+
+sealed class Ed25519PrivateKey : PrivateKey
+{
+    private const int _privateKeySize = 32;
```
Comment (tmds) [inline src/Tmds.Ssh/Ed25519PrivateKey.cs:12]: nit: for a `const`, the convention is `PrivateKeySize`.

**Comment by tmds** [comment]
Thanks for implementing this @jborean93!

File: src/Tmds.Ssh/Ed25519PrivateKey.cs (hunk 2 above)
Comment (jborean93) [reply src/Tmds.Ssh/Ed25519PrivateKey.cs]: Thanks I'll remember that for next time. Now that the array is already split in the parser it's no longer needed.

**Pull request merged**